executor/delete.go

```diff
@@ -49,12 +50,28 @@ func (e *DeleteExec) Next(ctx context.Context, req *chunk.Chunk) error {
 	return e.deleteSingleTableByChunk(ctx)
 }
 
-func (e *DeleteExec) deleteOneRow(tbl table.Table, handleIndex int, isExtraHandle bool, row []types.Datum) error {
+func (e *DeleteExec) deleteOneRow(tbl table.Table, handleIndex []int, isExtraHandle bool, row []types.Datum) error {
 	end := len(row)
 	if isExtraHandle {
 		end--
 	}
-	handle := kv.IntHandle(row[handleIndex].GetInt64())
+	var handle kv.Handle
+	if !tbl.Meta().IsCommonHandle {
+		handle = kv.IntHandle(row[handleIndex[0]].GetInt64())
+	} else {
+		pkDts := make([]types.Datum, 0, len(handleIndex))
+		for _, ordinal := range handleIndex {
+			pkDts = append(pkDts, row[ordinal])
+		}
+		handleBytes, err := codec.EncodeKey(e.ctx.GetSessionVars().StmtCtx, nil, pkDts...)
+		if err != nil {
+			return err
+		}
+		handle, err = kv.NewCommonHandle(handleBytes)
+		if err != nil {
+			return err
+		}
```

Comment:
This seems duplicate with the lines in `getKeysNeedCheckOneRow`. Can we extract them into 
```
buildHandleFromIndexes(...) kv.Handle
```
?